File: app/src/main/java/com/lin/meet/db_bean/ReplyHolder.java
package com.lin.meet.db_bean;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.lin.meet.R;
import com.lin.meet.override.ScaleAnim;

import de.hdodenhof.circleimageview.CircleImageView;

public class ReplyHolder extends RecyclerView.ViewHolder {
    public CircleImageView header;
    public LinearLayout item;
    public ImageView thumb;
    public TextView content;
    public TextView nickName;
    public TextView replyCount;
    public TextView thumbCount;
    public TextView time;
    public LinearLayout like;
    public ReplyHolder(@NonNull View itemView) {
        super(itemView);
        item = itemView.findViewById(R.id.comment_item);
        header = itemView.findViewById(R.id.comment_header);
        content = itemView.findViewById(R.id.comment_content);
        nickName = itemView.findViewById(R.id.comment_nickName);
        replyCount = itemView.findViewById(R.id.comment_reply);
        thumbCount = itemView.findViewById(R.id.thumb_count);
        time = itemView.findViewById(R.id.comment_time);
        thumb = itemView.findViewById(R.id.comment_thumb);
        like = item.findViewById(R.id.comment_like);
    }

    public void loadHeader(Context context,String uri){
        Glide.with(context).load(uri).into(header);
    }

    public void setLikeAnim(Boolean like){
        if(like){
            ScaleAnim.startAnim(thumb,R.mipmap.like2);
        }
        else {
            ScaleAnim.startAnim(thumb,R.mipmap.like);
        }
    }

    public void setLickCount(int count){
        if(count==0){
            thumbCount.setText("点赞");
        }
        else
            thumbCount.setText(String.valueOf(count));
    }

    public void setLike(boolean like){
        if(like){
            thumb.setImageResource(R.mipmap.like2);
        }else {
            thumb.setImageResource(R.mipmap.like);
        }
    }
}

File: app/src/main/java/com/lin/meet/drawer_message/view_holder/ReplyHolder.kt
package com.lin.meet.drawer_message.view_holder

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide
import com.lin.meet.R
import de.hdodenhof.circleimageview.CircleImageView

class ReplyHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    val nickname = itemView.findViewById<TextView>(R.id.nickName)
    val content = itemView.findViewById<TextView>(R.id.content)
    val replyContent = itemView.findViewById<TextView>(R.id.replyContent)
    val time = itemView.findViewById<TextView>(R.id.time)
    val header = itemView.findViewById<CircleImageView>(R.id.header)
    val img = itemView.findViewById<ImageView>(R.id.img)
    val item:View = itemView.findViewById(R.id.replyItem)
    fun setImgVisiable(show:Boolean){img.visibility = if (show) View.VISIBLE else View.GONE}
    fun loadHeader(uri:String){
        Glide.with(header.context).load(uri).into(header)}
    fun loadImg(uri:String){
        if(uri!="@null"&&uri.isNotEmpty()){
            setImgVisiable(true)
            Glide.with(img.context).load(uri).into(img)
        }else{
            setImgVisiable(false)
        }
    }
}